src/psyclone/tests/dynamo0p3_transformations_test.py

```diff
@@ -6616,40 +6616,123 @@ def test_intergrid_err(dist_mem):
         lftrans.apply(loops[0], loops[1])
     assert expected_err in str(excinfo)
 
+# Start OpenACC section
 
-def test_no_acc():
-    '''
-    Check that attempting to add any sort of OpenACC directive to a
-    dynamo0p3 InvokeSchedule causes an error.
 
-    '''
-    from psyclone.transformations import ACCEnterDataTrans, ACCLoopTrans, \
-        ACCParallelTrans
-    accdt = ACCEnterDataTrans()
-    accpt = ACCParallelTrans()
-    acclt = ACCLoopTrans()
+# Class ACCEnterDataTrans start
+def test_accenterdatatrans():
+    '''Test that an ACCEnterDataTrans transformation can add an OpenACC
+    Enter Data directive to the PSy-layer in the dynamo0.3 API.
 
+    '''
+    from psyclone.transformations import ACCEnterDataTrans
+    acc_enter_trans = ACCEnterDataTrans()
     _, info = parse(os.path.join(os.path.dirname(os.path.abspath(__file__)),
                                  "test_files", "dynamo0p3",
-                                 "1_single_invoke.f90"),
-                    api=TEST_API)
+                                 "1_single_invoke.f90"))
+    psy = PSyFactory(TEST_API, distributed_memory=False).create(info)
+    sched = psy.invokes.get('invoke_0_testkern_type').schedule
+    _ = acc_enter_trans.apply(sched)
+    code = str(psy.gen)
+    assert (
+        "      !$acc enter data\n"
+        "      !\n"
+        "      DO cell=1,f1_proxy%vspace%get_ncell()\n" in code)
+
+# Class ACCEnterDataTrans end
+
+
+# Class ACCKernelsTrans start
+def test_acckernelstrans():
+    '''
+    Test that an ACCKernelsTrans transformation can add an OpenACC
+    Kernels directive to the PSy layer in the dynamo0.3 API.
+
+    '''
+    from psyclone.transformations import ACCKernelsTrans
+    kernels_trans = ACCKernelsTrans()
+    _, info = parse(os.path.join(BASE_PATH, "1_single_invoke.f90"))
+    psy = PSyFactory(TEST_API, distributed_memory=False).create(info)
+    sched = psy.invokes.get('invoke_0_testkern_type').schedule
+    _ = kernels_trans.apply(sched.children)
+    code = str(psy.gen)
+    assert (
+        "      !$acc kernels\n"
+        "      DO cell=1,f1_proxy%vspace%get_ncell()\n" in code)
+    assert (
+        "      END DO \n"
+        "      !$acc end kernels\n" in code)
+
+# Class ACCKernelsTrans end
+
+
+# Class ACCParallelTrans start
+def test_accparalleltrans():
+    '''
+    Test that an ACCParallelTrans transformation can add an OpenACC
+    Parallel directive to the PSy layer in the dynamo0.3 API. An
+    EnterData directive is also required otherwise the transformation
```

Comment:
"otherwise an exception is raised at code-generation time." I think.

Reply:
Yes. Clarified.